mteb/models/rerankers_custom.py

```diff
@@ -0,0 +1,251 @@
+from __future__ import annotations
+
+import logging
+from functools import partial
+from typing import Any, Callable
+
+import torch
+from sentence_transformers import CrossEncoder
+from transformers import AutoModelForSequenceClassification, AutoTokenizer
+
+from mteb.encoder_interface import Encoder
+from mteb.evaluation.evaluators.RetrievalEvaluator import DenseRetrievalExactSearch
+from mteb.model_meta import ModelMeta
+
+logger = logging.getLogger(__name__)
+
+
+class RerankerWrapper(DenseRetrievalExactSearch):
+    def __init__(
+        self,
+        model_name_or_path: str,
+        batch_size: int = 4,
+        fp_options: bool = None,
+        silent: bool = False,
+    ):
+        self.model_name_or_path = model_name_or_path
+        self.batch_size = batch_size
+        self.fp_options = fp_options if fp_options is not None else torch.float32
+        if self.fp_options == "auto":
+            self.fp_options = torch.float32
+        elif self.fp_options == "float16":
+            self.fp_options = torch.float16
+        elif self.fp_options == "float32":
+            self.fp_options = torch.float32
+        elif self.fp_options == "bfloat16":
+            self.fp_options = torch.bfloat16
+        print(f"Using fp_options of {self.fp_options}")
+        self.device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+        self.silent = silent
+        self.first_print = True  # for debugging
+
+    def predict(self, input_to_rerank, **kwargs) -> list:
+        pass
+
+
+class BGEReranker(RerankerWrapper):
+    name: str = "BGE"
+
+    def __init__(
+        self,
+        model_name_or_path="BAAI/bge-reranker-v2-m3",
+        torch_compile=False,
+        **kwargs,
+    ):
+        super().__init__(model_name_or_path, **kwargs)
+        if not self.device:
+            self.device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+        model_args = {}
+        if self.fp_options:
+            model_args["torch_dtype"] = self.fp_options
+
+        try:
+            from FlagEmbedding import FlagReranker
+        except ImportError:
+            raise ImportError(
+                "FlagEmbedding is not installed. Please install it via `pip install FlagEmbedding`"
```

Comment:
```suggestion
                "FlagEmbedding is not installed. Please install it via `pip install "mteb[FlagEmbedding]"`"
```

To allows us to lock the dependencies to ensure that it works in the future. Naturally also add to the dependencies.